base/lock.jl

```diff
@@ -164,6 +164,17 @@ function lock(f, l::AbstractLock)
     end
 end
 
+struct Locked{L<:AbstractLock, T}
+    lock::L
+    value::T
+end
+
+function lock(f::Callable, l::Locked)
```

Comment:
```suggestion
function lock(f, l::Locked)
```

Since any Julia object can be a callable, I don't think `f` should be restricted to `Function`s and `Type`s.  Similar discussions: https://github.com/JuliaLang/julia/pull/34296 https://github.com/JuliaLang/julia/pull/30382

Reply:
I've removed the `::Callable` restriction.